crates/sparrow-runtime/src/metadata/raw_metadata.rs

```diff
@@ -246,36 +248,43 @@ impl RawMetadata {
 /// Arrow also does not support Decimal types. As of now, we are currently
 /// dropping the columns that are Decimal types since we do not support at query
 /// time either.
-fn convert_schema(schema: &Schema) -> SchemaRef {
+fn convert_schema(schema: &Schema) -> error_stack::Result<SchemaRef, Error> {
     let fields = schema
         .fields()
         .iter()
-        .filter_map(|field| {
-            match field.data_type() {
-                DataType::Timestamp(time_unit, Some(tz)) => {
-                    // TODO: We discard this because the conversion from an Arrow
-                    // schema to the Schema protobuf currently fails on such timestamp columns.
-                    info!(
-                        "Discarding time zone {:?} on timestamp column '{}'",
-                        tz,
-                        field.name()
-                    );
-                    Some(Field::new(
-                        field.name(),
-                        DataType::Timestamp(time_unit.clone(), None),
-                        field.is_nullable(),
-                    ))
-                }
-                DataType::Decimal128(_, _) | DataType::Decimal256(_, _) => {
-                    // TODO: Support decimal columns
-                    info!("Discarding decimal column '{}'", field.name());
-                    None
-                }
-                _ => Some(field.clone()),
-            }
-        })
-        .collect();
-    Arc::new(Schema::new(fields))
+        .map(|field| convert_field(field))
+        .collect::<Result<Vec<_>, _>>()?;
+    Ok(Arc::new(Schema::new(fields)))
+}
+
+/// TODO: Docs
+fn convert_field(field: &Field) -> error_stack::Result<Field, Error> {
+    match field.data_type() {
+        DataType::Timestamp(time_unit, Some(tz)) => {
+            // TODO: We discard this because the conversion from an Arrow
+            // schema to the Schema protobuf currently fails on such timestamp columns.
+            info!(
+                "Discarding time zone {:?} on timestamp column '{}'",
+                tz,
+                field.name()
+            );
```

Comment:
I'd make this `tracing::warn!("Time zones are unsupported. Interpreting column '{}' with time zone '{}' as UTC")`